postgresql/resource_postgresql_grant_test.go

```diff
@@ -583,18 +583,33 @@ func TestAccPostgresqlGrantObjects(t *testing.T) {
 				),
 			},
 			{
-				Config: fmt.Sprintf(testGrant, `["test_table", "test_table2"]`),
+				Config: fmt.Sprintf(testGrant, "test", `["test_table", "test_table2"]`, `["SELECT"]`),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("postgresql_grant.test", "objects.#", "2"),
 					resource.TestCheckResourceAttr("postgresql_grant.test", "objects.0", "test_table"),
 					resource.TestCheckResourceAttr("postgresql_grant.test", "objects.1", "test_table2"),
 					func(*terraform.State) error {
-						return testCheckTablesPrivileges(t, dbName, roleName, testTables, []string{"SELECT"})
+						return testCheckTablesPrivileges(t, dbName, roleName, testTables[0:2], []string{"SELECT"})
 					},
 				),
 			},
 			{
-				Config: fmt.Sprintf(testGrant, `["test_table", "test_view"]`),
+				// A separate resource with different privileges should not affect an unrelated resource
+				Config: fmt.Sprintf(testGrant, "test", `["test_table", "test_table2"]`, `["SELECT"]`) +
+					fmt.Sprintf(testGrant, "test2", `["test_table3"]`, `["SELECT", "UPDATE"]`),
```

Comment:
One thing I learned is if you want to introduce a new resource in a test step without deleting the resource from the previous step, you need to provide both. It made sense in retrospect because what you're supplying isn't necessarily a _resource_ but an entire _configuration_.

Reply:
That...actually makes a lotta sense. Good to know!